src/components/deviceprofilebuilder.js

```diff
@@ -1,279 +1,296 @@
-                    Container: "ts",
-                    Type: "Video",
-                    AudioCodec: hlsVideoAudioCodecs.join(","),
-                    VideoCodec: hlsVideoCodecs.join(","),
-                    Context: "Streaming",
-                    Protocol: "hls",
-                    MaxAudioChannels: physicalAudioChannels.toString(),
-                    MinSegments: "1",
-                    BreakOnNonKeyFrames: false
-                });
-            }
-
-            if (codecSupport.hasVP8Support() || codecSupport.hasVP9Support()) {
-                TranscodingProfiles.push({
-                    Container: "webm",
-                    Type: "Video",
-                    AudioCodec: "vorbis",
-                    VideoCodec: "vpx",
-                    Context: "Streaming",
-                    Protocol: "http",
-                    // If audio transcoding is needed, limit channels to number of physical audio channels
-                    // Trying to transcode to 5 channels when there are only 2 speakers generally does not sound good
-                    MaxAudioChannels: physicalAudioChannels.toString()
-                });
-            }
-        }
-
-        return TranscodingProfiles;
-    }
-
-    /**
-     * @returns {Array} Subtitle profiles.
-     */
-    function getSubtitleProfiles() {
-        //TODO: Add TTML support
-        let subProfiles = [];
-
-        if (codecSupport.hasTextTrackSupport(deviceId)) {
-            subProfiles.push({
-                Format: "vtt",
-                Method: "External"
-            });
-
-            subProfiles.push({
-                Format: "vtt",
-                Method: "Hls"
-            });
-        }
-
-        return subProfiles;
     }
 
-    /**
-     * Creates a device profile containing supported codecs for the active Cast device.
-     * @param {Object} [options=Object]
-     * @returns {Object} Device profile.
-     */
-    function getDeviceProfile(options = {}) {
-        profileOptions = options;
-        deviceId = castDevices.getActiveDeviceId();
-
-        const bitrateSetting = options.bitrateSetting || codecSupport.getMaxBitrateSupport();
-
-        //MaxStaticBitrate seems to be for offline sync only
-        let profile = {
-            MaxStreamingBitrate: 120000000, //bitrateSetting,
-            MaxStaticBitrate: 0,
-            MusicStreamingTranscodingBitrate: Math.min(bitrateSetting, 192000)
-        };
-
-        profile.DirectPlayProfiles = getDirectPlayProfiles();
-        profile.TranscodingProfiles = getTranscodingProfiles();
-        profile.ContainerProfile = getContainerProfiles();
-        profile.CodecProfiles = getCodecProfiles();
-        profile.SubtitleProfiles = getSubtitleProfiles();
-        profile.ResponseProfiles = getResponseProfiles();
-
-        return profile;
+    return TranscodingProfiles;
+}
+
+/**
+ * @returns {Array} Subtitle profiles.
+ */
+function getSubtitleProfiles() {
+    //TODO: Add TTML support
```

Comment:
Unrelated but spaces before comments are nice.

Reply:
I agree, I'll update those comments :p. Does it look alright aside from that?

Reply:
Yeah the conversion looks good.